modules/caddyhttp/caddyauth/command.go

```diff
@@ -67,17 +68,28 @@ func cmdHashPassword(fs caddycmd.Flags) (int, error) {
 	salt := []byte(fs.String("salt"))
 
 	if len(plaintext) == 0 {
-		if terminal.IsTerminal(int(os.Stdin.Fd())) {
-			fmt.Print("Enter password: ")
-			plaintext, err = terminal.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Println()
+		fd := int(os.Stdin.Fd())
+		if terminal.IsTerminal(fd) {
+			// ensure the terminal state is restored on SIGINT
+			state, _ := terminal.GetState(fd)
+			c := make(chan os.Signal)
+			signal.Notify(c, os.Interrupt)
+			go func() {
+				<-c
+				_ = terminal.Restore(fd, state)
+				os.Exit(caddy.ExitCodeFailedStartup)
+			}()
+
+			fmt.Fprint(os.Stderr, "Enter password: ")
```

Comment:
I think you misunderstood. If you do `caddy hash-password > file.txt` then it will contain:

```
Enter password: 
Confirm password: 
JDJhJDEwJGRCVzVGY0o1UHhZWDJqczRCdnF6aXVxTUppM0xaQmhUZDViUG9ZT0NsU0NvLmZhSDhHOFMu
```

That doesn't make sense, it should only contain the password. Also, if you use `>` then you won't see the prompts. It's better to print those on stderr.

Reply:
Ah :-/

This is unfortunate. Stderr is reserved for logs and errors.

Reply:
Well this is a special command, I don't think it's really that big of a problem, ultimately. I think usability here is more important :man_shrugging:

Reply:
FWIW, there's precedent for this: https://pymotw.com/3/getpass/index.html

That's a python lib, it specifically recommends using stderr if the stdout output is useful, which is the case here.